crates/sui-framework/sources/balance.move

```diff
@@ -42,8 +39,7 @@ module sui::balance {
     }
 
     /// Create a new supply for type T.
-    public fun create_supply<T: drop>(witness: T): Supply<T> {
-        assert!(sui::types::is_one_time_witness(witness), EBadWitness);
```

Comment:
BTW with this check removed, we would have to remove `coin::treasury_from_supply` in order to guarantee that all TreasuryCaps are made via one time witnesses

Reply:
> BTW with this check removed, we would have to remove `coin::treasury_from_supply` in order to guarantee that all TreasuryCaps are made via one time witnesses

FYI. This function does not seem to be used from any Sui Move code:
```
find ./ -name "*.move" ! -path '*/build/*' ! -path '*/deps/*' | xargs grep treasury_from_supply
.//crates/sui-framework/sources/coin.move:    public fun treasury_from_supply<T>(total_supply: Supply<T>, ctx: &mut TxContext): TreasuryCap<T> {

```

Are we OK removing it so that we can adopt @damirka's changes and land the whole thing, or we anticipate some important use cases here, in which case - what do we do? :-)

Reply:
Yeah I should add that we need to remove this conversion! It was there just in case it’s needed.